src/main/java/com/google/sps/servlets/RegistrationServlet.java

```diff
@@ -27,22 +34,50 @@
 @WebServlet("/registration")
 public class RegistrationServlet extends HttpServlet {
 
+  private UserService userService = UserServiceFactory.getUserService();
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String role = request.getParameter("role");
-    String name = request.getParameter("name");
-    String email = request.getParameter("email");
+    String role = getParameter(request, "role").orElse(null);
+    String firstName = getParameter(request, "first-name").orElse(null);
+    String lastName = getParameter(request, "last-name").orElse(null);
+    String fullName = firstName + " " + lastName;
```

Comment:
This will produce unexpected results if the previous parameters are null.
`null + "mystr" == "nullmystr"`

If these values are required, you can throw an exception if they are not present (such as IllegalArgumentException). If they are not required, a better default value may be the empty string ""

Reply:
I threw an exception using "orElseThrow" in one of my latest commits. Lmk how it looks!

Reply:
looks good!